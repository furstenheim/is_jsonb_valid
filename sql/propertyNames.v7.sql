-- propertyNames validation
-- all property names valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{"f":{},"foo":{}}');
-- some property names invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{"foo":{},"foobar":{}}');
-- object without properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{}');
-- ignores arrays
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '[1,2,3,4]');
-- ignores strings
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '"foobar"');
-- ignores other non-objects
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '12');
-- propertyNames validation with pattern
-- matching property names valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{"a":{},"aa":{},"aaa":{}}');
-- non-matching property name is invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{"aaA":{}}');
-- object without properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{}');
-- propertyNames with boolean schema true
-- object with any properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":true}', '{"foo":1}');
-- empty object is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":true}', '{}');
-- propertyNames with boolean schema false
-- object with any properties is invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":false}', '{"foo":1}');
-- empty object is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":false}', '{}');