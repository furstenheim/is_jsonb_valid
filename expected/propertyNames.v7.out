-- propertyNames validation
-- all property names valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{"f":{},"foo":{}}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- some property names invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{"foo":{},"foobar":{}}');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- object without properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- ignores arrays
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '[1,2,3,4]');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- ignores strings
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '"foobar"');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- ignores other non-objects
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"maxLength":3}}', '12');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- propertyNames validation with pattern
-- matching property names valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{"a":{},"aa":{},"aaa":{}}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- non-matching property name is invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{"aaA":{}}');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- object without properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":{"pattern":"^a+$"}}', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- propertyNames with boolean schema true
-- object with any properties is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":true}', '{"foo":1}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- empty object is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":true}', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

-- propertyNames with boolean schema false
-- object with any properties is invalid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":false}', '{"foo":1}');
 is_jsonb_valid_draft_v7 
-------------------------
 f
(1 row)

-- empty object is valid
SELECT is_jsonb_valid_draft_v7('{"propertyNames":false}', '{}');
 is_jsonb_valid_draft_v7 
-------------------------
 t
(1 row)

